book/src/tutorial/db.md

```diff
@@ -14,11 +14,11 @@ In `calc`, the database struct is in the [`db`] module, and it looks like this:
 ```
 
 The `#[salsa::db(...)]` attribute takes a list of all the jars to include.
-The struct must have a field named `storage` whose types is `salsa::Storage<Self>`, but it can also contain whatever other fields you want.
+The struct must have a field named `storage` whose type is `salsa::Storage<Self>`, but it can also contain whatever other fields you want.
 The `storage` struct owns all the data for the jars listed in the `db` attribute.
 
 The `salsa::db` attribute autogenerates a bunch of impls for things like the `salsa::HasJar<crate::Jar>` trait that we saw earlier.
-This means that
+This means that ... [what goes here?]
```

Comment:
This sentence just abruptly ends; what should go here?

Reply:
I have no idea :) let's just remove it?